lib/daru/io/importers/linkages/sql.rb

```diff
@@ -0,0 +1,27 @@
+require 'daru'
+
+module Daru
+  class DataFrame
+    class << self
+      # Read a database query and returns a Dataset
+      #
+      # @param dbh [DBI::DatabaseHandle, String] A DBI connection OR Path to a SQlite3 database.
+      # @param query [String] The query to be executed
+      #
+      # @return A dataframe containing the data resulting from the query
+      #
+      # USE:
+      #
+      #  dbh = DBI.connect("DBI:Mysql:database:localhost", "user", "password")
+      #  Daru::DataFrame.from_sql(dbh, "SELECT * FROM test")
+      #
+      #  #Alternatively
+      #
+      #  require 'dbi'
+      #  Daru::DataFrame.from_sql("path/to/sqlite.db", "SELECT * FROM test")
+      def from_sql(dbh, query)
+        Daru::IO::Importers::SQL.load dbh, query
+      end
+    end
+  end
+end
```

Comment:
I had a better look at meta-programming, and it feels like we can automate these linkages with `define_singleton_method`. 

```ruby
#! daru/io/importers/linkages.rb
module Daru
  class DataFrame
    class << self
      def register_importer(function, instance)
        define_singleton_method(function) { |*args| instance.new(*args).call }
      end

      def register_all_importers
        importers = Daru::IO::Importers
        klasses   = importers.constants.select {|c| importers.const_get(c).is_a? Class}
        klasses.each do |klass|
          method_name = "from_#{klass.to_s.downcase}".to_sym
          register_importer method_name, Object.const_get("Daru::IO::Importers::#{klass}")
        end     
      end
    end
  end
end

Daru::DataFrame.register_all_importers
```

The best part about this approach is that, it works well with partial requires too. However, one minor-ish drawback is that when there are multiple libraries for same format, they should be called like `Daru::DataFrame#from_fastestcsv` (not `#csv`)which may or may not be intuitive to users.

@zverok @v0dro @lokeshh - Please share your thoughts regarding whether we can continue forward with this. 😄